pkg/apis/garden/validation/validation.go

```diff
@@ -2216,6 +2217,39 @@ func validateKubeProxy(kp *garden.KubeProxyConfig, fldPath *field.Path) field.Er
 	return allErrs
 }
 
+func validateMonitoring(monitoring *garden.Monitoring, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if monitoring.Alerting != nil {
+		allErrs = append(allErrs, validateAlerting(monitoring.Alerting, fldPath.Child("alerting"))...)
+	}
+	return allErrs
+}
+
+func validateAlerting(alerting *garden.Alerting, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	temp := []string{}
+	for _, email := range alerting.EmailReceivers {
+		if !utils.TestEmail(email) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("emailReceivers"), alerting.EmailReceivers, "must provide a valid email"))
+		}
+		if contains(temp, email) {
+			allErrs = append(allErrs, field.Duplicate(fldPath.Child("emailReceivers"), alerting.EmailReceivers))
+		} else {
+			temp = append(temp, email)
+		}
+	}
+	return allErrs
+}
+
+func contains(s []string, element string) bool {
```

Comment:
Why don't you use a `map[string]struct{}` or `sets.String` to check for duplicates? Then you don't have to implement this `contains` function and the code above is easier to read.